### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by ewjoachim**
This PR consists of 3 main parts:

- Boilerplate
- The README which consists of pretty much all the user documentation for the extension
- The code & tests of the extension itself

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (mgu) [inline docs/Makefile:19]: missing newline ?

File: sphinx_github_changelog/__init__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+def setup(app):
+    token_name = "sphinx_github_changelog_token"
+    app.add_config_value(
+        name=token_name, default=os.environ.get(token_name.upper()), rebuild="html"
+    )
+    app.add_directive("changelog", changelog.ChangelogDirective)
+
+    return {
+        "version": "0.1",
```
Comment (mgu) [inline sphinx_github_changelog/__init__.py:25]:
```suggestion
        "version": __version__,
```

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,63 @@
+[tox]
+envlist =
+    {
+    py 37,py 38,py36}-tests,check-lint,docs
```
Comment (mgu) [inline tox.ini:4]: strange formatting (and spaces) :)

File: docs/Makefile (hunk 1 above)
Comment (ewjoachim) [reply docs/Makefile:19]: Hm, this file was generated a long time ago by sphinx, then copied over to my cookiecutter and then generated. I think I never actually read it :D

File: sphinx_github_changelog/__init__.py (hunk 2 above)
Comment (ewjoachim) [reply sphinx_github_changelog/__init__.py:25]: Indeed. This wasn’t used before but now that’s it’s a full fledged extension, we can set this properly

File: tox.ini (hunk 3 above)
Comment (ewjoachim) [reply tox.ini:4]: Hm, might have to fix a thing or two in the cookiecutter...

Comment (ewjoachim) [reply tox.ini:4]: A logical consequence is that .travis.yml is equally broken.

**Comment by ewjoachim** [comment]
@mgu :+1: ?

**Pull request merged**